src/main/java/com/sandvoxel/generitech/common/tileentities/power/TestPower.java

```diff
@@ -26,195 +27,44 @@
-
-
-            if (te1 instanceof TileEntityMachineBase && getTransfer(pos.up()) > 100 && container.getStoredPower() != 0) {
-
-                if (te instanceof ITeslaHolder) {
-                    if (((ITeslaHolder) te).getStoredPower() > 100) {
-                        if (!worldObj.isRemote) {
-                            GeneriTech.network.sendToServer(new PacketPower(T0transfer, pos.getX(), pos.getZ(), pos.getY() + 1));
-                        }
-                        this.container.takePower(T0transfer, false);
-                    } else {
-                        if (!worldObj.isRemote) {
-                            GeneriTech.network.sendToServer(new PacketPower(((ITeslaHolder) te).getStoredPower(), pos.getX(), pos.getZ(), pos.getY() + 1));
-                        }
-                        this.container.takePower(((ITeslaHolder) te).getStoredPower(), false);
-                    }
-                }
-            } else {
-                if (te1 instanceof TileEntityMachineBase && container.getStoredPower() != 0) {
-                    if (!worldObj.isRemote) {
-                        GeneriTech.network.sendToServer(new PacketPower(getTransfer(pos.down()), pos.getX(), pos.getZ(), pos.getY() + -1));
-                    }
-                    this.container.takePower(getTransfer(pos.down()), false);
-                }
-
-            }
-
-
-            if (te instanceof ITeslaHolder && te1 instanceof ITeslaHolder && te1 instanceof TileEntityMachineBase == false) {
-                long abbs = ((ITeslaHolder) te).getStoredPower() - ((ITeslaHolder) te1).getStoredPower();
-                //System.out.println(abbs);
-                if (abbs > 100) {
-                    if (!worldObj.isRemote) {
-                        GeneriTech.network.sendToServer(new PacketPower(T0transfer, pos.getX(), pos.getZ(), pos.getY() + 1));
-
-                    }
-                    this.container.takePower(T0transfer, false);
-                } else {
-                    if (te1 instanceof ITeslaConsumer && abbs != 0 && abbs / 2 != 0) {
-                        if (!worldObj.isRemote)
-
-                            GeneriTech.network.sendToServer(new PacketPower(abbs / 2, pos.getX(), pos.getZ(), pos.getY() + 1));
-                    }
-                    this.container.takePower(abbs / 2, false);
-                    if (te1 instanceof ITeslaConsumer && abbs == 1) {
-                        {
-                            System.out.println(abbs);
-
-                            if (!worldObj.isRemote) {
-                                GeneriTech.network.sendToServer(new PacketPower(1, pos.getX(), pos.getZ(), pos.getY() + 1));
-                            }
-                            this.container.takePower(1, false);
-                        }
-                    }
-                }
-            }
-        }
 
+        BlockPos pos = getPos();
+        World wold = getWorld();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "update" is 48 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)